### ISSUE TITLE
Screen reader repeats last word twice at end of line and on empty lines with Ctrl+Arrow

### ISSUE THREAD

**Issue by snufas** [microsoft/vscode#248606]
Type: <b>Bug</b>

## Description:
When using a screen reader in Visual Studio Code, moving the cursor word by word with Ctrl+Right Arrow causes two issues:
1. The last word in a non-empty line is repeated twice—once when the cursor is at the start of the word, and again when it’s at the end of the line.
2. When moving onto empty lines, the screen reader keeps repeating the last word from the previous non-empty line instead of announcing “blank” or staying silent.

## Example file:
```
apple orange
(empty line)
(empty line)
```

## Steps to reproduce:
1. Place the cursor at the start of the word “apple.”
2. Press Ctrl+Right Arrow four times.

## The issue:
- First press: screen reader says “orange” (cursor is at the start of “orange”)
- Second press: screen reader says “orange” again (cursor is at the end of “orange”)
- Third press: moves to first empty line, screen reader says “orange”
- Fourth press: moves to second empty line, screen reader says “orange”

So, “orange” is read out loud four times—twice while still on the non-empty line, then twice more on the empty lines.

## Expected behavior:
Using the same example file, starting from the word “apple” and pressing Ctrl+Right Arrow four times, the screen reader should announce:
- “orange” (cursor is at the start of “orange”)
- “end of line” (cursor is at the end of “orange”)
- “blank” (first empty line)
- “blank” (second empty line)

## Note:
This issue has been around for a while. I just noticed it now, but I tested and confirmed it still happens in much older versions of Visual Studio Code.

Tested on:
- Visual Studio Code development version
  Version: 1.91.0-insider (system setup)
  Commit: 0da59bdde2339d6ba632524dc8f3fb6f4b8a2bcb
  Date: 2024-06-14T10:07:53.538Z
  Electron: 29.4.0
  ElectronBuildId: 9593362
  Chromium: 122.0.6261.156
  Node.js: 20.9.0
  V8: 12.2.281.27-electron.0
  OS: Windows_NT x64 10.0.19045

VS Code version: Code 1.100.0 (19e0f9e681ecb8e5c09d8784acaa601316ca4571, 2025-05-07T12:48:53.763Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz (12 x 3498)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.91GB (18.53GB free)|
|Process Argv|C:\\Users\\Vytautas\\OneDrive\\Desktop\\DEV\\python\\Think_Python\\Think_Python book\\208_Exercise.txt --crash-reporter-id 7c280da0-24a3-4763-9e9f-4345e9e5c40f|
|Screen Reader|yes|
|VM|33%|
</details><details><summary>Extensions (39)</summary>

Extension|Author (truncated)|Version
---|---|---
path-intellisense|chr|2.10.0
paste-indented|cii|1.5.0
dbclient-jdbc|cwe|1.4.4
vscode-myssql-client2|cwe|8.2.9
add-to-git-ignore|djl|1.0.1
docker|doc|0.6.0
hscopes|dra|0.0.9
git-stage|dun|1.0.1
gitlens|eam|17.0.3
prettier-vscode|esb|11.0.0
vscode-firefox-debug|fir|2.15.0
copilot|Git|1.317.0
copilot-chat|Git|0.27.0
metalevel-references|let|0.1.2
indentation-level-movement2|Mik|1.3.0
vscode-docker|ms-|1.29.6
vscode-edge-devtools|ms-|2.1.8
black-formatter|ms-|2025.2.0
debugpy|ms-|2025.8.0
isort|ms-|2025.0.0
python|ms-|2025.6.0
vscode-pylance|ms-|2025.4.1
jupyter|ms-|2025.4.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.1.0
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.413.0
remote-ssh|ms-|0.120.0
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.99.0
remote-explorer|ms-|0.5.0
vscode-speech|ms-|0.14.0
vsliveshare|ms-|1.0.5948
text-power-tools|qcz|1.51.0
vscode-xml|red|0.29.0
multi-command|ryu|1.6.0
code-spell-checker|str|4.0.47
vscodeintellicode|Vis|1.3.2

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vscod805cf:30301675
binariesv615:30325510
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dwnewjupytercf:31046870
pythonrstrctxt:31112756
nativeloc1:31192215
5fd0e150:31155592
dwcopilot:31170013
6074i472:31201624
dwoutputs:31242946
customenabled:31248079
hdaa2157:31222309
copilot_t_ci:31222730
e5gg6876:31282496
pythoneinst12:31285622
bgtreat:31268568
4gafe986:31271826
31787653:31262186
3e8i5726:31271747
996jf627:31283433
pythonrdcb7cf:31303019
usemplatestapi:31297334
7bj51361:31289155
747dc170:31275177
g20af354:31278749
aj953862:31281341
generatesymbolt:31295002
convertfstringf:31295003
gendocf:31295004

```

</details>

**Comment by snufas** [comment]
@meganrogge I kind of forgot to mention that I tested this with Jaws, NVDA, and Narrator. The same  issue exists. I have a suspicion it's a known issue because it exists for a whilebased on the results of my old version of Visual Studio Code Development version. Even if it might be a known issue, I figured I need to submit that anyways, because it needs fixing. And my suspicion is it could be a part of electron problem rather than a VS code, but either way the solution needs to be implemented at some point and looked at for sure.

**Comment by SeanTolstoyevski** [comment]
Hello all and @meganrogge ,

I solved this issue upstream. Chromium should no longer merge line endings with the previous word / punch.
If the VSCode team produces an Electron.js runtime with the latest version of Chromium this issue will be resolved. Or we will wait for the electron team to update to the   upstream version.

The Chrome CL is here: [Accessibility: Fix caret navigation over newlines in IA2 word boundary (7319018) · Gerrit Code Review](https://chromium-review.googlesource.com/c/chromium/src/+/7319018)

Sory for duplicate comment.
